### ISSUE TITLE
Update feature switches documentation

### ISSUE THREAD

**Issue by LakshanF** [dotnet/docs#41739]
### Describe the issue or suggestion

Ensure that the feature switches [documentation](https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/trimming-options#trimming-framework-library-features) is up to date with the recently added new ones. Also, it would be good to have the `AppContext` settings for each build property.

**Comment by MichalStrehovsky** [comment]
Cc @eerhardt @dotnet/illink I don't know how may of these are actually implementation details. The repo docs in the dotnet/runtime repo are not public docs, they're docs on implementation details. E.g. I'm not sure about scenarios where we'd want to support/document the appcontext switches.

**Comment by eerhardt** [comment]
I think there's a different level of "public" between things like the runtime and higher-level app models (like MAUI and WinForms), versus "public" that we want .NET application developers to be aware of.

A decent amount of these feature switches should never be used by .NET application developers, but they were created for specific app model needs. For example `DisableDependencyInjectionDynamicEngine` was created for Android, but could be useful for other app models in the future. These kinds of switches feel like they shouldn't be given the same level of publicity as things like `InvariantGlobalization`, which can be specified by application devs.

I think the switches that are meant for .NET application developers should be documented on learn.microsoft.com. The switches that aren't, I don't think they need to be.

Of the above list, ones that I think application developers should be aware of:

* PredefinedCulturesOnly
* JsonSerializerIsReflectionEnabledByDefault (maybe? cc @eiriktsarpalis for thoughts)
* HybridGlobalization (probably? cc @ilonatommy for thoughts on whether app devs should know about this)
* DebuggerSupport
* EventSourceSupport
* UseNativeHttpHandler
* UseSystemResourceKeys

**Comment by ilonatommy** [comment]
@eerhardt, you are correct, we want to make sure developers are aware of `HybridGlobalization` on WASM. We have it in our [docs](https://github.com/dotnet/runtime/blob/8a28b8c334b95b47165ea941579a067ab613dc5a/docs/design/features/globalization-hybrid-mode.md) but adding it here as well would be helpful.
